### PR TITLE
Reworks stasis, reshuffles life order, makes gases use defines.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Mainly a port of https://github.com/Baystation12/Baystation12/pull/20114/files for the time being, except for sleeper changes.

Todo:

- [ ] Test if life isn't fucked for simple mobs.

Myazaki -- Note to explain what this does for the testmerge:
- Instead of being binary (on/off), stasis now has stasis levels. e.g. A Cryo-tube will provide stasis from 0 to 20 to its occupant depending on temperature.
- Stasis x20 means that you will tick Life() 20 times less than normal, so you will die slower, heal slower. But these things *will* happen, whereas the previous system simply turned them off.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!wip

File: maps/exodus/exodus-1_station.dmm [hunk 1]
```diff
@@ -64111,7 +64111,7 @@
 /obj/machinery/computer/general_air_control{
 	frequency = 1441;
 	name = "Tank Monitor";
-	sensors = list("n2_sensor" = "Nitrogen", "o2_sensor" = "Oxygen", "co2_sensor" = "Carbon Dioxide", "tox_sensor" = "Toxins", "n2o_sensor" = "Nitrous Oxide", "waste_sensor" = "Gas Mix Tank")
+	sensors = list("n2_sensor" = GAS_NITROGEN, "o2_sensor" = GAS_OXYGEN, "co2_sensor" = "Carbon Dioxide", "tox_sensor" = "Toxins", "n2o_sensor" = "Nitrous Oxide", "waste_sensor" = "Gas Mix Tank")
```
Comment (NonQueueingMatt) [inline maps/exodus/exodus-1_station.dmm:64114]:
```suggestion
	sensors = list("n2_sensor" = "nitrogen", "o2_sensor" = "oxygen", "co2_sensor" = "Carbon Dioxide", "tox_sensor" = "Toxins", "n2o_sensor" = "Nitrous Oxide", "waste_sensor" = "Gas Mix Tank")
```

File: maps/aurora/aurora-4_mainlevel.dmm [hunk 2]
```diff
@@ -14049,7 +14049,7 @@
 /obj/machinery/computer/general_air_control{
 	frequency = 1441;
 	name = "Tank Monitor";
-	sensors = list("n2_sensor" = "Nitrogen", "o2_sensor" = "Oxygen", "co2_sensor" = "Carbon Dioxide", "ph_sensor" = "Phoron", "air_sensor" = "Air Tank")
+	sensors = list("n2_sensor" = GAS_NITROGEN, "o2_sensor" = GAS_OXYGEN, "co2_sensor" = "Carbon Dioxide", "ph_sensor" = "Phoron", "air_sensor" = "Air Tank")
```
Comment (NonQueueingMatt) [inline maps/aurora/aurora-4_mainlevel.dmm:14052]:
```suggestion
	sensors = list("n2_sensor" = "nitrogen", "o2_sensor" = "oxygen", "co2_sensor" = "Carbon Dioxide", "ph_sensor" = "Phoron", "air_sensor" = "Air Tank")
```

File: code/modules/research/xenoarchaeology/tools/suspension_generator.dm [hunk 3]
```diff
@@ -247,7 +247,7 @@
 			for(var/mob/living/carbon/C in T)
 				C.weakened += 5
 				C.visible_message("<span class='notice'>\icon[C] [C] begins to float in the air!","You feel tingly and light, but it is difficult to move.</span>")
-		if("nitrogen")
+		if(GAS_NITROGEN)
```
Comment (NonQueueingMatt) [inline code/modules/research/xenoarchaeology/tools/suspension_generator.dm:250]:
```suggestion
		if("nitrogen")
```

File: code/modules/research/xenoarchaeology/tools/suspension_generator.dm [hunk 4]
```diff
@@ -84,7 +84,7 @@
 	if(!locked)
 		dat += "<b>Select field mode</b><br>"
 		dat += "[field_type=="carbon"?"<b>":""			]<A href='?src=\ref[src];select_field=carbon'>Diffracted carbon dioxide laser</A></b><br>"
-		dat += "[field_type=="nitrogen"?"<b>":""		]<A href='?src=\ref[src];select_field=nitrogen'>Nitrogen tracer field</A></b><br>"
+		dat += "[field_type==GAS_NITROGEN?"<b>":""		]<A href='?src=\ref[src];select_field=nitrogen'>Nitrogen tracer field</A></b><br>"
```
Comment (NonQueueingMatt) [inline code/modules/research/xenoarchaeology/tools/suspension_generator.dm:87]:
```suggestion
		dat += "[field_type=="nitrogen"?"<b>":""		]<A href='?src=\ref[src];select_field=nitrogen'>Nitrogen tracer field</A></b><br>"
```

File: code/modules/research/xenoarchaeology/finds/finds_defines.dm [hunk 5]
```diff
@@ -83,7 +83,7 @@
 		if(ARCHAEO_PEN)
 			return "mercury"
 		if(ARCHAEO_CRYSTAL)
-			return "nitrogen"
+			return GAS_NITROGEN
```
Comment (NonQueueingMatt) [inline code/modules/research/xenoarchaeology/finds/finds_defines.dm:86]:
```suggestion
			return "nitrogen"
```

File: code/modules/research/xenoarchaeology/finds/finds_defines.dm [hunk 6]
```diff
@@ -93,9 +93,9 @@
 		if(ARCHAEO_CULTROBES)
 			return "potassium"
 		if(ARCHAEO_SOULSTONE)
-			return "nitrogen"
+			return GAS_NITROGEN
```
Comment (NonQueueingMatt) [inline code/modules/research/xenoarchaeology/finds/finds_defines.dm:96]:
```suggestion
			return "nitrogen"
```

File: code/modules/research/xenoarchaeology/finds/finds_defines.dm [hunk 7]
```diff
@@ -93,9 +93,9 @@
...
 		if(ARCHAEO_SHARD)
-			return "nitrogen"
+			return GAS_NITROGEN
```
Comment (NonQueueingMatt) [inline code/modules/research/xenoarchaeology/finds/finds_defines.dm:98]:
```suggestion
			return "nitrogen"
```

File: code/modules/random_map/noise/ore.dm [hunk 8]
```diff
@@ -61,22 +61,22 @@
 				T.resources["silver"] =   rand(RESOURCE_LOW_MIN,  RESOURCE_LOW_MAX)
 				T.resources["uranium"] =  rand(RESOURCE_LOW_MIN,  RESOURCE_LOW_MAX)
 				T.resources["diamond"] =  0
-				T.resources["phoron"] =   0
+				T.resources[GAS_PHORON] =   0
```
Comment (mikomyazaki) [inline code/modules/random_map/noise/ore.dm:64]:
```suggestion
				T.resources[MATERIAL_PHORON] =   0
```

File: code/modules/random_map/noise/ore.dm [hunk 9]
```diff
@@ -61,22 +61,22 @@
...
+				T.resources[GAS_PHORON] =   rand(RESOURCE_LOW_MIN,  RESOURCE_LOW_MAX)
 				T.resources["osmium"] =   rand(RESOURCE_MID_MIN,  RESOURCE_MID_MAX)
 				T.resources["hydrogen"] = 0
 				T.resources["diamond"] =  0
 				T.resources["iron"] =     0
 			else                             // Deep metals.
 				T.resources["uranium"] =  rand(RESOURCE_LOW_MIN,  RESOURCE_LOW_MAX)
 				T.resources["diamond"] =  rand(RESOURCE_LOW_MIN,  RESOURCE_LOW_MAX)
-				T.resources["phoron"] =   rand(RESOURCE_LOW_MIN, RESOURCE_LOW_MAX)
+				T.resources[GAS_PHORON] =   rand(RESOURCE_LOW_MIN, RESOURCE_LOW_MAX)
```
Comment (mikomyazaki) [inline code/modules/random_map/noise/ore.dm:79]:
```suggestion
				T.resources[MATERIAL_PHORON] =   rand(RESOURCE_LOW_MIN, RESOURCE_LOW_MAX)
```

File: code/modules/mining/ore_datum.dm [hunk 10]
```diff
@@ -119,7 +119,7 @@ var/global/list/ore_data = list()
 	spread_chance = 5
 	ore = /obj/item/ore/diamond
 	scan_icon = "mineral_rare"
-	xarch_source_mineral = "nitrogen"
+	xarch_source_mineral = GAS_NITROGEN
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/modules/mining/ore_datum.dm:122]: I have no fucking clue

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/machinery/cryo.dm [hunk 11]
```diff
@@ -106,16 +106,17 @@
 		return
 
 	// this is the data which will be sent to the ui
-	var/data[0]
+	var/list/data = list()
 	data["isOperating"] = on
 	data["hasOccupant"] = occupant ? 1 : 0
 
-	var/occupantData[0]
+	var/kist/occupantData = list()
```
Comment (mikomyazaki) [inline code/game/machinery/cryo.dm]:
```suggestion
	var/list/occupantData = list()
```

File: maps/exodus/exodus-1_station.dmm [hunk 12]
```diff
@@ -64105,7 +64105,7 @@
 /obj/machinery/computer/general_air_control{
 	frequency = 1441;
 	name = "Tank Monitor";
-	sensors = list("n2_sensor" = "Nitrogen", "o2_sensor" = "Oxygen", "co2_sensor" = "Carbon Dioxide", "tox_sensor" = "Toxins", "n2o_sensor" = "Nitrous Oxide", "waste_sensor" = "Gas Mix Tank")
+	sensors = list("n2_sensor" = "nitrogen", "o2_sensor" = "oxygen", "co2_sensor" = "Carbon Dioxide", "tox_sensor" = "Toxins", "n2o_sensor" = "Nitrous Oxide", "waste_sensor" = "Gas Mix Tank")
```
Comment (NonQueueingMatt) [inline maps/exodus/exodus-1_station.dmm:64108]:
```suggestion
	sensors = list("n2_sensor" = "Nitrogen", "o2_sensor" = "Oxygen", "co2_sensor" = "Carbon Dioxide", "tox_sensor" = "Toxins", "n2o_sensor" = "Nitrous Oxide", "waste_sensor" = "Gas Mix Tank")
```

File: maps/aurora/aurora-4_mainlevel.dmm [hunk 13]
```diff
@@ -14053,7 +14053,7 @@
 /obj/machinery/computer/general_air_control{
 	frequency = 1441;
 	name = "Tank Monitor";
-	sensors = list("n2_sensor" = "Nitrogen", "o2_sensor" = "Oxygen", "co2_sensor" = "Carbon Dioxide", "ph_sensor" = "Phoron", "air_sensor" = "Air Tank")
+	sensors = list("n2_sensor" = "nitrogen", "o2_sensor" = "oxygen", "co2_sensor" = "Carbon Dioxide", "ph_sensor" = "Phoron", "air_sensor" = "Air Tank")
```
Comment (NonQueueingMatt) [inline maps/aurora/aurora-4_mainlevel.dmm:14056]:
```suggestion
	sensors = list("n2_sensor" = "Nitrogen", "o2_sensor" = "Oxygen", "co2_sensor" = "Carbon Dioxide", "ph_sensor" = "Phoron", "air_sensor" = "Air Tank")
```

**Pull request merged**